vaadin-button-flow-parent/vaadin-button-flow/src/test/java/com/vaadin/flow/component/button/tests/ButtonTest.java

```diff
@@ -300,29 +314,19 @@ public void disableOnClick_click_disablesComponent() {
         Assert.assertFalse(
                 "Button should have been disabled when event has been fired",
                 buttonIsEnabled.get());
-    }
-
-    @Test
-    public void disableOnClick_serverRevertsDisabled_stateChangesAdded() {
-        button = new Button();
-        button.setDisableOnClick(true);
-
-        button.click();
 
         StateNode node = button.getElement().getNode();
         HashMap<String, Serializable> changeTracker = node.getChangeTracker(
                 node.getFeature(ElementAttributeMap.class), () -> null);
-        Assert.assertEquals(
-                "Change should have been set for disabled attribute", "true",
+        Assert.assertEquals("Disabled attribute should be set to true", "true",
                 changeTracker.get("disabled"));
-        Assert.assertFalse("Button should be disabled", button.isEnabled());
-
-        changeTracker.clear();
-
-        button.addClickListener(event -> event.getSource().setEnabled(true));
+    }
 
+    @Test
+    public void disableOnClick_clickRevertsDisabled_componentIsEnabled() {
```

Comment:
**note:** The previous version of the test attempted to click the disabled button and that expectedly stopped working after the PR change.